google/resource_compute_instance.go

```diff
@@ -964,6 +981,55 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 		d.SetPartial("tags")
 	}
 
+	if d.HasChange("instance_state") {
+		if d.Get("instance_state") == "running" {
+			_, err := config.clientCompute.Instances.Start(project,
+				zone, d.Id()).Do()
+			if err != nil {
+				return fmt.Errorf(
+					"Error starting an instance (%s): %s", d.Id(), err)
+			}
+
+			stateConf := &resource.StateChangeConf{
```

Comment:
We can probably extract this whole chunk into a function like `waitForTerminalState` and wait for either terminal status?